suitest.js

```diff
@@ -0,0 +1,209 @@
+const clearAppDataFactory = require('./lib/chains/clearAppDataChain');
+const cookieFactory = require('./lib/chains/cookieChain');
+const sleepFactory = require('./lib/chains/sleepChain');
+const pressButtonFactory = require('./lib/chains/pressButtonChain');
+const positionFactory = require('./lib/chains/positionChain');
+const windowFactory = require('./lib/chains/windowChain');
+const executeCommandFactory = require('./lib/chains/executeCommandChain');
+const runTestFactory = require('./lib/chains/runTestChain');
+// const executeBrightScriptFactory = require('./lib/chains/executeBrightScriptChain');
+const jsExpressionFactory = require('./lib/chains/javascriptExpressionChain');
+// const brightScriptExpressionFactory = require('./lib/chains/brightScriptExpressionChain');
+const elementFactory = require('./lib/chains/elementChain');
+const pollUrlFactory = require('./lib/chains/pollUrlChain');
+const networkRequestFactory = require('./lib/chains/networkRequestChain');
+const videoFactory = require('./lib/chains/videoChain');
+const playstationVideoFactory = require('./lib/chains/playstationVideoChain');
+
+// Constants
+const {ELEMENT_PROP, VALUE} = require('./lib/constants/element');
+const {PROP_COMPARATOR} = require('./lib/constants/comparator');
+const VIDEO_STATE = require('./lib/constants/videoState');
+const CONTENT_MODE = require('./lib/constants/contentMode');
+const ELEMENT_STATE = require('./lib/constants/elementState');
+const TEXT_ALIGNMENT = require('./lib/constants/textAlignment');
+const IMAGE_LOAD_STATE = require('./lib/constants/imageLoadState');
+const BORDER_STYLE = require('./lib/constants/borderStyle');
+const VISIBILITY_STATE = require('./lib/constants/visibilityState');
+const VRC = require('./lib/constants/vrc');
+const KEY = require('./lib/constants/keys');
+const {NETWORK_PROP, NETWORK_METHOD} = require('./lib/constants/networkRequest');
+const HAD_NO_ERROR = require('./lib/constants/hadNoError');
+
+// Network
+const webSocketsFactory = require('./lib/api/webSockets');
+const unusedExpressionWatchersFactory = require('./lib/utils/unusedExpressionWatchers');
+const ipcClient = require('./lib/testLauncher/ipc/client');
+const ipcServer = require('./lib/testLauncher/ipc/server');
+
+// Contexts
+const Context = require('./lib/utils/Context');
+const AuthContext = require('./lib/utils/AuthContext');
+const {configFactory} = require('./config');
+const {setUpRaven} = require('./lib/utils/sentry/Raven');
+const {warnLauncherAndLibHasDiffVersions} = require('./lib/utils/packageMetadataHelper');
+const {createLogger} = require('./lib/utils/logger');
+
+// Publicly available API goes here
+class Suitest {
+	constructor({exitOnError}) {
+		// instance dependencies
+		this.authContext = new AuthContext();
+		this.appContext = new Context();
+		this.pairedDeviceContext = new Context();
+		this.testContext = new Context();
+		this.configuration = configFactory();
+		this.config = this.configuration.config;
```

Comment:
Not sure if this one should be editable. Perhaps freeze it somehow.

I would make `config` a getter in this class, so that it's always returning `this.configuration.config`. Then, I would freeze `config` in config module and re-set it to a new instance each time it's overriden.